src/scrypt.rs

```diff
@@ -32,8 +32,8 @@ pub fn derive_key(pass: &Password, salt: &[u8; 32], n: u32, p: u32, r: u32) -> R
     }
 
     let mut derived_key = vec![0u8; KEY_LENGTH];
-    let scrypt_params = ScryptParams::new(log_n, r, p);
-    scrypt(pass.as_bytes(), salt, &scrypt_params, &mut derived_key);
+    let scrypt_params = ScryptParams::new(log_n, r, p)?;
```

Comment:
IMO, this change needs a test to check whether the result is the same as the previous implementation.

Reply:
I added a [test](https://github.com/CodeChain-io/rust-codechain-crypto/pull/8/commits/a687270eaca77140177d4cd8683f725919518448#r375666923).